File: companies/Paytm/176.second-highest-salary.java
/**
 * LeetCode 176 is a SQL problem, so there is no Java runtime API. This Java
 * file preserves the canonical SQL answer and the algorithm explanation.
 *
 * Algorithm:
 * Find the maximum salary strictly below the overall maximum. MAX over an
 * empty set returns NULL, which is the required result when no second salary
 * exists.
 *
 * Complexity:
 * Logical scan time O(n), aggregate space O(1).
 */
class Solution {
    static final String SOLUTION_SQL = """
        SELECT MAX(salary) AS SecondHighestSalary
        FROM Employee
        WHERE salary < (SELECT MAX(salary) FROM Employee);
        """;
}

File: companies/Paytm/176.second-highest-salary.go
package leetcode

// SolutionSQL176 is the SQL answer for LeetCode 176. The source file is Python
// only as a repository container for a SQL problem.
//
// Explanation: MAX(salary) below the global maximum returns the second-highest
// distinct salary. If no such salary exists, SQL MAX over an empty set returns
// NULL, matching the required output.
const SolutionSQL176 = `SELECT MAX(salary) AS SecondHighestSalary
FROM Employee
WHERE salary < (SELECT MAX(salary) FROM Employee);`
